modules/effects/src/main/scala/dev/profunktor/redis4cats/effects.scala

```diff
@@ -208,4 +207,29 @@ object effects {
     /** Set expiry only when the new expiry is greater than current one */
     case object Lt extends ExpireExistenceArg
   }
+
+  sealed abstract class RedisType(val asString: String)
+  object RedisType {
+    def fromString(s: String): Option[RedisType] = s match {
+      case "string" => Some(String)
+      case "list"   => Some(List)
+      case "set"    => Some(Set)
+      case "zset"   => Some(SortedSet)
+      case "hash"   => Some(Hash)
+      case "stream" => Some(Stream)
+      case _        => None
+    }
+    case object String extends RedisType("string")
+
+    case object List extends RedisType("list")
+
+    case object Set extends RedisType("set")
+
+    case object SortedSet extends RedisType("zset")
+
+    case object Hash extends RedisType("hash")
+
+    case object Stream extends RedisType("stream")
```

Comment:
I don't have much experience with **Redis** but any reason for not including other types `Bitmap`, `Bitfield`, and `Geospatial`?
Also, I wonder if **Json**: https://redis.io/docs/latest/develop/data-types/json/ is its own type?

Reply:
I believe redis types refers to their primitives types , which is just the list I provided. Everything else is implemented in terms of those types.
I will doublecheck and showcase it in the tests also